pkg/kubectl/cmd/util/factory.go

```diff
@@ -1224,3 +1224,20 @@ func (f *Factory) NewBuilder(thirdPartyDiscovery bool) *resource.Builder {
 
 	return resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true))
 }
+
+func (f *Factory) ResourcesWithPodTemplates() []*meta.RESTMapping {
```

Comment:
@smarterclayton @knobunc is this the best name for this function? In order to be modified a pod has to have a template but it doesn't always have one.